components/bl-pdfViewer-component/src/index.js

```diff
@@ -0,0 +1,105 @@
+import { Document, Page } from './react-pdf.min.js';
+import { useState, useEffect, useRef } from 'react';
+import { Controls, NoData } from './subcomponents';
+
+const { cn } = BackendlessUI.CSSUtils;
+
+export default function PdfViewer({ component, eventHandlers }) {
+  const { style, display, classList, pdfUrl, renderType, width, height } = component;
+  const { onLoadSuccess, onLoadError } = eventHandlers;
+
+  const [numPages, setNumPages] = useState(null);
+  const [pageIndex, setPageIndex] = useState(1);
+  const [documentRef, setDocumentRef] = useState();
+  const [controlsDisabled, setControlsDisabled] = useState(false);
+
+  const inputRef = useRef();
+
+  useEffect(() => {
+    inputRef.current.style.width = ((inputRef.current.value.length + 1) * 9) + 'px';
```

Comment:
We could make an input transparent and for 4 digits with right content align - that will looks good, and could be written without JS